Fix variable-overwritten-before-usage reported on overriding default values in IF (#961)
MarketSquare/robotframework-robocop: docs/releasenotes/unreleased/fixes.4.rst

> variable-overwritten-before-usage should not be raised if overridden in an IF block (#950)

Reviewer: ```suggestion
variable-overwritten-before-usage should not be raised if overwritten in an IF block (#950)
```

------------------------------------------------------------------------------------------

W0922 ``variable-overwritten-before-usage`` warns if the variable was overwritten before first use. It should now

> ignore cases where the variable was initiated and then conditionally overriden in a IF block:

Reviewer: ```suggestion
ignore cases where the variable was initiated and then conditionally overwritten in an IF block:
```

```

> Conditionally Overriden Variable

Reviewer: ```suggestion
Conditionally Overwritten Variable
```
